src/main/java/org/terasology/navgraph/BaseRegion.java

```diff
@@ -20,30 +20,52 @@
 
 /**
  * @author synopia
+ * Abstract class which floors and regions derive from
  */
 public abstract class BaseRegion<N extends BaseRegion> {
     public int id;
     public Set<N> neighborRegions = new HashSet<N>();
     protected BitMap map;
     protected Floor floor;
 
+    /**
+     * Creates a BaseRegion
+     * @param id the id of the base region
+     */
     protected BaseRegion(int id) {
         this.id = id;
         map = new BitMap();
     }
 
+    /**
+     * Marks a point in the region as walkable by characters
+     * @param x the x co-ordinate of the point with respect to region's top-left corner
+     * @param y the y co-ordinate of the point with respect ot region's rop-left corner
+     */
     public void setPassable(int x, int y) {
         map.setPassable(x, y);
     }
 
+    /**
+     * Function to get the map representing the region
+     * @return bitmap of the region
+     */
     public BitMap getMap() {
         return map;
     }
 
+    /**
+     * Returns neighbours with whom this region shares a side
+     * @return set of neighbouring regions
+     */
     public Set<N> getNeighborRegions() {
         return neighborRegions;
     }
 
+    /**
+     *
+     * @return the id of the base region along with the number of neighbouring regions
+     */
     @Override
     public String toString() {
         return id + "\nrc = " + neighborRegions.size();
```

Comment:
If not strictly necessary I'd think that the string representation could simply be something like `"NavRegion(<id>, <size>)"`, avoiding line breaks.